class-udm-updater.php

```diff
@@ -97,6 +102,28 @@ public function __construct($mothership, $muid, $relative_plugin_file, $options
 		}
 	}
 
+	/**
+	 * Disconnect a cloned site by unsetting the `email` option and resetting PUC update cache and at the same time adding `site_host_path` as a plugin option for a cloned site checking purpose in future
+	 */
+	public function disconnect_cloned_site() {
+
+		$udm_options = $this->get_option($this->option_name);
+		$site_url = parse_url(network_site_url());
+		$site_host_path = isset($site_url['host']) ? $site_url['host'] : '';
+		$site_host_path .= isset($site_url['path']) ? $site_url['path'] : '';
+		/**
+		 * If it's a cloned site, other plugins that use the same site ID may have still connected. Just unset the email and clear the PUC update cache so that when the users reconnect they get the warning message about the duplicate site ID issue
+		 * any releases of UDM that doesn't have the `site_host_path` option must first have their `email` option unset and PUC update cache reset, including any plugin that was cloned from a site where the plugin is connected but it has a different URL site compared to the site_host_path option
+		 */
+		if (empty($udm_options['site_host_path']) || (!empty($udm_options['email']) && strtolower($udm_options['site_host_path']) !== strtolower($site_host_path))) {
+			// Since we're not disconnecting the plugin from the updates server, doing this on both cloned and non-cloned sites won't make the entitlement inactive. However, the user will be able to connect again without getting any issue even the updates server finds the plugin is still connected (active)
+			unset($udm_options['email']);
+			$udm_options['site_host_path'] = $site_host_path;
+			$this->plug_updatechecker->resetUpdateState();
+		}
+		$this->update_option($this->option_name, $udm_options);
```

Comment:
@andtown This is going to mean that an SQL `UPDATE` command (which possibly changes nothing) gets run on every visit to the relevant dashboard pages, doesn't it?

Reply:
@DavidAnderson684 Yes, you're right, It's going to query the database every time the relevant pages get visited even if there are no changes were made. I think it'd be nice if it could be called only when needed. I've now corrected this.